src/Spice86/Program.cs

```diff
@@ -34,37 +30,9 @@ public static void RunWithOverrides<T>(string[] args, string expectedChecksum) w
     /// </summary>
     /// <param name="args">The command-line arguments.</param>
     [STAThread]
-    public static void Main(string[] args)
-    {
-        Configuration configuration = CommandLineParser.ParseCommandLine(args);
-
-        if (!configuration.HeadlessMode)
-        {
-            OxyPlotModule.EnsureLoaded();
-            BuildAvaloniaApp().StartWithClassicDesktopLifetime(args, Avalonia.Controls.ShutdownMode.OnMainWindowClose);
-        }
-        else
-        {
-            ServiceProvider serviceProvider = Startup.StartupInjectedServices(args);
-            ILoggerService? loggerService = serviceProvider.GetService<ILoggerService>();
-            if (loggerService is null)
-            {
-                throw new InvalidOperationException("Could not get logging service from DI !");
-            }
-
-            ProgramExecutor programExecutor = new ProgramExecutor(loggerService, null, configuration);
-            programExecutor.Run();
-        }
-    }
-
-    /// <summary>
-    /// Configures and builds an Avalonia application instance.
-    /// </summary>
-    /// <returns>The built <see cref="AppBuilder"/> instance.</returns>
-    public static AppBuilder BuildAvaloniaApp()
-    {
-        return AppBuilder.Configure<App>()
-            .UsePlatformDetect()
-            .LogToTrace();
+    public static void Main(string[] args) {
+        using TopLevelContainer topLevelContainer = new();
+        Owned<Startup> startup = topLevelContainer.Resolve();
+        startup.Value.StartApp(args);
```

Comment:
I'm still reading the StrongInject docs, but should this not be
```suggestion
        topLevelContainer.Run(startup => startup.Start(args));
```
?